File: LeetCode/0455. Assign Cookies/solution.cpp
/**
 * 21 / 21 test cases passed.
 * Status: Accepted
 * Runtime: 80 ms
 */
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int i = 0, j = 0;
        while (i < g.size() && j < s.size()) {
            if (g[i] <= s[j]) i++;
            j++;
        }
        return i;
    }
};

File: LeetCode/0455. Assign Cookies/solution.py
"""
21 / 21 test cases passed.
Status: Accepted
Runtime: 80 ms
"""
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        candy = 0
        ans = 0
        for child in g:
            while candy < len(s) and s[candy] < child:
                candy += 1
            
            if candy < len(s):
                ans += 1
                candy += 1
        return ans
